### DISCUSSION TITLE
Nullish with Default

### DISCUSSION THREAD

**Discussion by stychu** [colinhacks/zod#2179] [Q&A] [answered]
Given this schema

```ts
import { z } from 'zod';

const status = z
  .object({
    currentStatus: z.string().default(''),
    closureReason: z.string().default(''),
  })

const schema = z.object({
  status: status.default(status).nullish().default(status),
});


const a = schema.parse({
  status: null
})
```

I was expecting that when I do `  status: null` I will get back my default but I get null. Works for undefined tho 👀 . Am I missing something ?

**Comment by JacobWeisenburger** [comment, marked as answer]
https://github.com/colinhacks/zod#default

> Conceptually, this is how Zod processes default values:
>
> 1. If the input is `undefined`, the default value is returned
> 2. Otherwise, the data is parsed using the base schema

**Reply by davideicardi** [reply]
There is a way to consider `null` values as `undefined`? So that we can apply a default and in general "normalize" values?
I have tried something like this:
```
// Return undefined if null
function NullAsUndefined<T>(schema: z.ZodType<T>) {
  return z.preprocess((arg) => {
    if (arg === null) {
      return undefined;
    }
    return arg;
  }, schema);
}
```
Do you see problems?
Thank you!

**Reply by lunelson** [reply]
You could create a different `.default()` method by using `.transform()`

```ts
import { z } from 'zod'

const zDefaultString = z
  .string()
  .nullish()
  .transform((s) => s ?? '')

zDefaultString.parse(null) // ''
zDefaultString.parse(undefined) // ''
zDefaultString.parse('what') // 'what'
zDefaultString.parse('') // ''
```
